lib/shared/puppet_self_service.rb

```diff
@@ -46,4 +46,72 @@ def self.service_running_enabled(name, service = nil)
 
     service_running(name, service) and service_enabled(name, service)
   end
+
+  # Return active postgres resource in Debian
+  # @return [String] pe-postgres fully qualified service is debian
+  def self.pe_postgres_version
+    if Facter.value(:os)['family'].eql?('Debian')
+      "pe-postgresql#{Facter.value(:pe_postgresql_info)['installed_server_version']}"
+    else
+      'pe-postgresql'
+    end
+  end
+
+  #  Checks if passed service file exists in correct directory for the OS
+  # @return [Boolean] true if file exists
+
+  def self.service_file_exist?(configfile)
+    configdir = if Facter.value(:os)['family'].eql?('RedHat') || Facter.value(:os)['family'].eql?('Suse')
+                  '/etc/sysconfig'
+                else
+                  '/etc/default'
+                end
+    File.exist?("#{configdir}/#{configfile}")
+  end
+
+  # Check if Primary node
+  # @return [Boolean] true is primary node
+  def self.primary?
+    service_file_exist?('pe-puppetserver') &&
+      service_file_exist?('pe-orchestration-services') &&
+      service_file_exist?('pe-console-services') &&
+      service_file_exist?('pe-puppetdb')
+  end
+
+  # Check if replica node
+  # @return [Boolean]
+  def self.replica?
+    service_file_exist?('pe-puppetserver') &&
+      !service_file_exist?('pe-orchestration-services') &&
+      service_file_exist?('pe-console-services') &&
+      service_file_exist?('pe-puppetdb')
+  end
+
+  # Check if Compiler node
+  # @return [Boolean]
+  def self.compiler?
+    service_file_exist?('pe-puppetserver') &&
+      !service_file_exist?('pe-orchestration-services') &&
+      !service_file_exist?('pe-console-services') &&
+      service_file_exist?('pe-puppetdb')
+  end
+
+  # Check if lagacy compiler node
+  # @return [Boolean] true
+  def self.legacy_compiler?
+    service_file_exist?('pe-puppetserver') &&
+      !service_file_exist?('pe-orchestration-services') &&
+      !service_file_exist?('pe-console-services') &&
+      !service_file_exist?('pe-puppetdb')
+  end
+
+  # Check if Pe postgres  node
+  # @return [Boolean]
+
+  def self.postgres?
```

Comment:
Shouldn't this check have a check for the `pe_postgres_version` file?

Reply:
It could, but the only node where pe:build is populated and all of those things are not present is a postgres node at this time

Reply:
Since we are being so explicit on the others, it seems like it is worthwhile here. My concern is mainly over false detection, which is not a bad thing with the current facts. If more facts rely on this to run `psql` commands or something of the like, it may be worth adding a test for `/opt/puppetlabs/server/bin/psql` to ensure it is a Postgres node.

Reply:
probably do this /etc/sysconfig/pe-pgsql/pe-postgresql